python/setup.py.in

```diff
@@ -51,13 +81,13 @@ def mkl():
     commit = git_commit()
     with open(filename, 'w') as f:
         f.write(cnt % {
-            'major': MAJOR,
-            'minor': MINOR,
-            'patch': PATCH,
+            'major': get_major(),
+            'minor': get_minor(),
+            'patch': get_patch(),
```

Comment:
Can there be any document? Perhaps we need to properly test it...
1. How is this generated?
2. What is generated?

Reply:
Can you add some comments for your codes here?